tests/config/test_search.py

```diff
@@ -9,119 +9,170 @@
-        aval_mock.side_effect = [True]
-        expected = os.path.expanduser('~/.atkinson/my_config.yml')
-        assert search.get_config_file(filename='my_config.yml') == expected
+        file_name = 'my_config.yml'
+        expected = [os.path.join(os.path.expanduser('~/.atkinson'), file_name)]
+        check = CheckAvailMock(expected)
+        aval_mock.side_effect = check.check
+        assert list(search.get_config_files(filenames=file_name)) == expected
 
 
 def test_custom_returned_slot_2():
     """Test config returned from /etc/atkinson"""
     with patch('atkinson.config.search._check_available') as aval_mock:
-        aval_mock.side_effect = [False, True]
-        assert search.get_config_file(filename='my_config.yml') == '/etc/atkinson/my_config.yml'
+        file_name = 'my_config.yml'
+        expected = [os.path.join('/etc/atkinson', file_name)]
+        check = CheckAvailMock(expected)
+        aval_mock.side_effect = check.check
+        assert list(search.get_config_files(filenames=file_name)) == expected
 
 
 def test_custom_returned_slot_3():
-    """Test config returned from ./config"""
+    """Test config returned from ./configs"""
     with patch('atkinson.config.search._check_available') as aval_mock:
-        aval_mock.side_effect = [False, False, True]
-        expected = os.path.realpath('./configs/my_config.yml')
-        assert search.get_config_file(filename='my_config.yml') == expected
+        file_name = 'my_config.yml'
+        expected = [os.path.join(os.path.realpath('./configs'), file_name)]
+        check = CheckAvailMock(expected)
+        aval_mock.side_effect = check.check
+        assert list(search.get_config_files(filenames=file_name)) == expected
 
 
 def test_default_override():
     """Test default config from override"""
     with patch('atkinson.config.search._check_available') as aval_mock:
-        aval_mock.side_effect = [True]
-        expected = os.path.expanduser('~/my_config_dir/config.yml')
-        assert search.get_config_file(overrides='~/my_config_dir') == expected
+        file_name = 'config.yml'
+        override = '~/my_config_dir'
+        expected = [os.path.join(os.path.expanduser(override), file_name)]
+        check = CheckAvailMock(expected)
+        aval_mock.side_effect = check.check
+        assert list(search.get_config_files(overrides=override)) == expected
 
 
 def test_default_override_not_found():
     """Test an override is given but the file is not found. Fall back to the defaults."""
     with patch('atkinson.config.search._check_available') as aval_mock:
-        aval_mock.side_effect = [False, True]
-        expected = os.path.expanduser('~/.atkinson/config.yml')
-        assert search.get_config_file(overrides=['~/my_config_dir']) == expected
+        override = ['~/my_config_dir']
+        file_name = 'config.yml'
+        expected = []
+        for my_dir in override:
+            expected.append(os.path.join(os.path.expanduser(my_dir), file_name))
```

Comment:
There is nothing dynamic here, why not simply:
`expected = [os.path.join(os.path.expanduser(my_dir), 'config.yml)]`

I think it would also make the test easier to read.